datalake_api/fetcher.py

```diff
@@ -84,7 +84,8 @@ def _get_s3_key(self, file_id):
         try:
             return self.s3_bucket.Object(path).get()
         except BotoClienError as e:
-            if e.response['ResponseMetadata']['HTTPStatusCode'] == 404:
+            if 'ResponseMetadata' in e.response and \
+                    e.response['ResponseMetadata']['HTTPStatusCode'] == 404:
                 msg = 'No file with id {} exists'.format(file_id)
```

Comment:
Did you encounter a case where ResponseMetadata was not present? If so, maybe there's a boto library difference we should know more about. Specifically, if there's another way to get a 404, we should accommodate it.